### PR TITLE
Added an SFU Copy-Paste Demo

### PR DESCRIPTION
**Pull request by stazio**
I'm back! To get back into the groove with all of your wonderful additions to the Track API (which you have done a great job with), I wrote an SFU demo to send video back to the browser. This demo takes a browser video source, changes the SSRC to [4](https://xkcd.com/221/), and sends it to an unlimited number of receivers.

### CODE REVIEW

**Comment by paullouisageneau** [comment]
Hey, nice to have you back, really cool demo!

File: examples/sfu-media/main.cpp [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+#include <nlohmann/json.hpp>
+
+using nlohmann::json;
+
+class Sender {
+    rtc::PeerConnection conn;
+};
+
+struct Receiver {
+    // TODO @paul.
```
Comment (paullouisageneau) [inline examples/sfu-media/main.cpp:36]: What do you mean here? :thinking:

File: examples/sfu-media/main.cpp [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+		auto session = std::make_shared<rtc::RtcpSession>();
+		track->setRtcpHandler(session);
+
+		const rtc::SSRC targetSSRC = 4;
+
+		track->onMessage(
+		    [&receivers](rtc::binary message) {
+			    // This is an RTP packet
+			    auto rtp = (rtc::RTP*) message.data();
+			    rtp->ssrc = htonl(targetSSRC);
```
Comment (paullouisageneau) [inline examples/sfu-media/main.cpp:75]: Do you really need to expose back all the RTP/RTCP structures in `rtp.h` just to set this? It's bit overkill, don't you think? :smile:

File: include/rtc/description.hpp [hunk 3]
```diff
@@ -71,6 +71,8 @@ class Description {
 		Direction direction() const { return mDirection; }
 		void setDirection(Direction dir);
 
+		void addSSRC(uint32_t ssrc, std::string name);
```
Comment (paullouisageneau) [inline include/rtc/description.hpp:74]: Actually `Entry` is the common class for application m-line and media m-line to help factoring what can be in the SDP parsing and generation, so media-only properties would be better in `Media`.

File: src/sctptransport.cpp [hunk 4]
```diff
@@ -717,7 +717,7 @@ int SctpTransport::RecvCallback(struct socket *sock, union sctp_sockstore addr,
 	return ret;
 }
 
-int SctpTransport::SendCallback(struct socket *sock, uint32_t sb_free) {
+int SctpTransport::SendCallback(struct socket *sock, uint32_t sb_free, void *ulp_info) {
```
Comment (paullouisageneau) [inline src/sctptransport.cpp:720]: You should not need to add this if you use the `usrsctp` version from `git submodule update`. Actually, I need to upgrade `usrsctp` and do some tests.

**Comment by stazio** [comment]
It's good to be back! I spend a lot of time changing RTP headers in my own code which is mostly why I brought it back out. Honestly, this file really should be it's own library (or be replaced by one).
I will also point you to the [https://github.com/stazio/libdatachannel/tree/api_updates](api_updates) branch that uses the RTP headers a *little* bit more.

**Comment by paullouisageneau** [comment]
@stazio Do you need a hand on this? I think merging master and cherry-picking your fixes from #176 would be sufficient to have something ready to merge, right?

**Comment by stazio** [comment]
Hi Paul, sorry about the delay. I think this API is more or less stable by this point. I am currently actively developing my product with #176 right now, and should, *hopefully* have it done enough to be able to mark these changes as stable and then I would love to fix the SFU demo and merge it tomorrow

**Comment by paullouisageneau** [comment]
Cool, great news!

**Comment by stazio** [comment]
Superseded by #176.

**Pull request closed without merging**